File: task2.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = scanner.nextInt();
        
        ArrayList<Integer> nums = new ArrayList<>();
        System.out.print("Enter the elements of the array: ");
        for (int i = 0; i < n; i++) {
            nums.add(scanner.nextInt());
        }
        
        int count = 0;
        for (int num : nums) {
            if (isOddDigits(num)) {
                count++;
            }
        }
        
        System.out.println("The number of elements with odd number of digits is: " + count);
    }
    
    public static boolean isOddDigits(int num) {
        int count = 0;
        while (num != 0) {
            count++;
            num /= 10;
        }
        return count % 2 != 0;
    }
}

File: task2.kt
import java.util.Scanner

fun isOddDigits(num: Int): Boolean {
    var count = 0
    var number = num
    while (number != 0) {
        count++
        number /= 10
    }
    return count % 2 != 0
}

fun main() {
    val scanner = Scanner(System.`in`)
    print("Enter the number of elements in the array: ")
    val n = scanner.nextInt()
    val nums = IntArray(n)
    print("Enter the elements of the array: ")
    for (i in 0 until n) {
        nums[i] = scanner.nextInt()
    }
    var count = 0
    for (num in nums) {
        if (isOddDigits(num)) {
            count++
        }
    }
    println("The number of elements with odd number of digits is: $count")
}
